drivers/adc/adc_rpi_pico.c

```diff
@@ -0,0 +1,352 @@
+/**
+ * @file
+ *
+ * @brief RaspberryPi Pico ADC driver
+ */
+
+/*
+ * Copyright 2021 Google LLC
+ * Copyright 2022 TOKITA Hiroshi <tokita.hiroshi@fujitsu.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT raspberrypi_pico_adc
+
+#include <drivers/adc.h>
+#include <kernel.h>
+#include <logging/log.h>
+#include <sys/byteorder.h>
+#include <sys/util.h>
+#include <zephyr.h>
+
+/*
+ * "hardware/adc.h cause conflict of adc_read() definition.
```

Comment:
I also ran into this when working on ADC long ago. The approach that was taken for the last conflict (the KHZ and MHZ macros) was to patch `hal_rpi_pico` (see https://github.com/zephyrproject-rtos/hal_rpi_pico/commit/191f5ba46fda49523cdaaef27583d1c875ba2c36).
Some of the functions of the ADC API are useless to us (like `adc_init`, which is unaware of pinctrl), but some may be useful. If using these functions would reduce code duplication, I am for using those instead of reimplementing them.

Reply:
I have issued
https://github.com/zephyrproject-rtos/hal_rpi_pico/pull/2
to avoid conflict between PICO-SDK and Zephyr.

I will update this PR after the changes are approved.

Reply:
Replace code by PICO-SDK that can replace with PICO-SDK's function.
If available no function that we expected, I implemented it with a similar interface to PICO-SDK's function.